trunk/includes/pardot-settings-class.php

```diff
@@ -1458,37 +1293,22 @@ static function get_setting( $key ) {
 	 *
 	 * @since 1.0.0
 	 */
-	static function set_setting( $key, $value ) {
+	static function set_setting($key, $value)
+	{
 		/**
-		 * Gran the array of settings
+		 * Grab the array of settings
 		 */
 		$settings = self::get_settings();
 
 		/**
 		 * Assign the setting for this key its value
 		 */
-		$settings[ $key ] = $value;
-
-        if ($settings['password'] != NULL) {
-            $settings['password'] = self::pardot_encrypt($settings['password'], true);
-        }
-
-        if ($settings['client_secret'] != NULL) {
-            $settings['client_secret'] = self::pardot_encrypt($settings['client_secret'], true);
-        }
-
-        if ($settings['api_key'] != NULL) {
-            $settings['api_key'] = self::pardot_encrypt($settings['api_key'], true);
-        }
-
-        if ($settings['refresh_token'] != NULL) {
-            $settings['refresh_token'] = self::pardot_encrypt($settings['refresh_token'], true);
-        }
+		$settings[$key] = $value;
 
```

Comment:
So this took me a while to figure out, but I believe that this was causing the bug with the campaign dropdown not showing up immediately after authentication.  At the end of this function we call `update_option`.  This subsequently calls `sanitize_fields`, which cleans the fields in various ways, then tries to authenticate.  However, if we encrypt the tokens first, then it won't be able to authenticate.  So now, we will encrypt the credentials after we sanitize with the `pre_update_option_pardot_settings` function.